kolibri/plugins/facility/assets/src/views/users/sidePanels/AssignCoachesSidePanel.vue

```diff
@@ -1,35 +1,347 @@
+  import { bulkUserManagementStrings } from 'kolibri-common/strings/bulkUserManagementStrings';
+  import { UserKinds } from 'kolibri/constants';
+  import RoleResource from 'kolibri-common/apiResources/RoleResource';
+  import useSnackbar from 'kolibri/composables/useSnackbar';
+  import commonCoreStrings, { coreStrings } from 'kolibri/uiText/commonCoreStrings';
+  import FacilityUserResource from 'kolibri-common/apiResources/FacilityUserResource';
+  import SelectableList from '../../common/SelectableList.vue';
+  import { _userState } from '../../../modules/mappers';
 
   export default {
     name: 'AssignCoachesSidePanel',
     components: {
       SidePanelModal,
+      KIcon,
+      SelectableList,
     },
     mixins: [commonCoreStrings],
+    setup(props) {
+      const selectedClasses = ref([]); // Array of selected class IDs
+      const isLoading = ref(false);
+      const showErrorWarning = ref(false);
+      const showUndoModal = ref(false);
+      const showCloseConfirmationModal = ref(false);
+      const createdRoles = ref([]);
+      const instance = getCurrentInstance();
+      const facilityUsers = ref([]);
+
+      const {
+        undoAssignCoachHeading$,
+        undoAssignCoachMessage$,
+        coachesAssignedNotice$,
+        assignCoachUndoneNotice$,
+        undoAction$,
+        usersInClassNotAffected$,
+        assignAction$,
+        searchForAClass$,
+        defaultErrorMessage$,
+        discardAction$,
+        discardWarning$,
+        keepEditingAction$,
+        disgardChanges$,
+        numUsersNotEligible$,
+        selectClassesLabel$,
+        assignUsersHeading$,
+        assignToAllClasses$,
+      } = bulkUserManagementStrings;
+      const { createSnackbar } = useSnackbar();
+      const { dismissAction$ } = coreStrings;
+
+      const loadUsers = async () => {
+        isLoading.value = true;
+        const users = await FacilityUserResource.fetchCollection({
+          getParams: {
+            by_ids: Array.from(props.selectedUsers).join(','),
+          },
+        });
+        facilityUsers.value = users.map(_userState);
+        isLoading.value = false;
+      };
+      loadUsers();
+
+      // Computed properties
+      const formattedClasses = computed(() => {
+        return [...props.classes]
+          .sort((a, b) => a.name.localeCompare(b.name))
+          .map(({ id, name }) => ({ id, label: name }));
+      });
+
+      const selectedUsersCount = computed(() => props.selectedUsers.size);
+
+      const hasSelectedClasses = computed(() => selectedClasses.value.length > 0);
+
+      const getSelectedClassObjects = computed(() =>
+        props.classes.filter(cls => selectedClasses.value.includes(cls.id)),
+      );
+
+      // Filter eligible users (coaches, admins, superusers)
+      const eligibleUsers = computed(() => {
+        return facilityUsers.value.filter(
+          user =>
+            props.selectedUsers.has(user.id) &&
```

Comment:
Now that `facilityUsers` are the selected users loaded from the api, this condition is no longer necessary, removing it will make that this condition is more readable.